src/common/hist_util.cu

```diff
@@ -320,15 +347,21 @@ HistogramCuts DeviceSketch(int device, DMatrix* dmat, int max_bins,
   dmat->Info().weights_.SetDevice(device);
   for (const auto& batch : dmat->GetBatches<SparsePage>()) {
     size_t batch_nnz = batch.data.Size();
-    for (auto begin = 0ull; begin < batch_nnz;
-         begin += sketch_batch_num_elements) {
+    auto const& info = dmat->Info();
+    dh::caching_device_vector<uint32_t> groups(info.group_ptr_.size());
+    thrust::copy(info.group_ptr_.cbegin(), info.group_ptr_.cend(), groups.begin());
```

Comment:
I think you can use the begin/end pointers directly in the vector constructor for groups to save one line.

Reply:
Thanks for the suggestion, applied.